### PR TITLE
Hw9 task2

### PR DESCRIPTION
**Pull request by E1-Bart0**

### CODE REVIEW

File: test/hw9/test_task2.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+
+def test_generator_suppressor__catch_expected_exception():
+    with Suppressor(IndexError):
+        [][2]
+    assert True
+
+
+def test_generator_suppressor__catch_not_expected_exception():
+    with pytest.raises(IndexError, match="list index out of range"):  # noqa: PT012
```
Comment (Cjkjvfnby) [inline test/hw9/test_task2.py:23]: What is PT012?

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

File: test/hw9/test_task2.py (hunk 1 above)
Comment (E1-Bart0) [reply test/hw9/test_task2.py:23]: PT012 pytest.raises() block should contain a single simple statement

Comment (E1-Bart0) [reply test/hw9/test_task2.py:23]: But I have 2 lines.

File: test/hw9/test_task2.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+def test_class_suppressor__catch_not_expected_exception():
+    with pytest.raises(IndexError, match="list index out of range"):  # noqa: PT012
+        with Suppressor(ZeroDivisionError):
+            [][2]
+
+
+def test_generator_suppressor__catch_expected_exception():
+    with Suppressor(IndexError):
+        [][2]
+    assert True
```
Comment (Cjkjvfnby) [inline test/hw9/test_task2.py]: Why do you need this?

**Review by Cjkjvfnby — APPROVED** [review]

**Pull request merged**